### PR TITLE
Add antlr error listener.

### PR DESCRIPTION
**Pull request by Genius-pig**
Add an antlr4 error listener. Display syntax error information in a client.

```
IoTDB> insert into root.ln.wf02.wt02(timestamp,status) value(1,true)
insert into root.ln.wf02.wt02(timestamp,status) value(1,true)
Msg: line 1:48 mismatched input 'value' expecting VALUES
It costs 0.445s
```

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/service/TSServiceImpl.java [hunk 1]
```diff
@@ -572,7 +573,11 @@ public TSExecuteStatementResp executeStatement(TSExecuteStatementReq req) {
       } else {
         return executeUpdateStatement(physicalPlan);
       }
-    } catch (SQLParserException e) {
+    } catch (ParseCancellationException e) {
+      logger.debug(e.getMessage());
+      return getTSExecuteStatementResp(getStatus(TSStatusCode.SQL_PARSE_ERROR, e.getMessage()));
+    }
+      catch (SQLParserException e) {
```
Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/service/TSServiceImpl.java]: format

**Comment by liutaohua** [comment]
Hello, could you help me check whether the server can return the normal prompt when I type the abnormal character abc? @Genius-pig

**Comment by Genius-pig** [comment]
> Hello, could you help me check whether the server can return the normal prompt when I type the abnormal character abc? @Genius-pig

I can't understand. Can you give me an example?

**Comment by liutaohua** [comment]
> > Hello, could you help me check whether the server can return the normal prompt when I type the abnormal character abc? @Genius-pig
> 
> I can't understand. Can you give me an example?
```
IoTDB> abc
abc
Msg: Fail to execute abc after reconnecting. please check server status
It costs 3.271s
```

**Comment by Genius-pig** [comment]
> > > Hello, could you help me check whether the server can return the normal prompt when I type the abnormal character abc? @Genius-pig
> > 
> > 
> > I can't understand. Can you give me an example?
> 
> ```
> IoTDB> abc
> abc
> Msg: Fail to execute abc after reconnecting. please check server status
> It costs 3.271s
> ```

```
IoTDB> abc
abc
Msg: line 1:0 mismatched input 'abc' expecting {CREATE, INSERT, UPDATE, DELETE, SELECT, SHOW, GRANT, SET, DESCRIBE, UNLINK, ADD, LINK, DROP, MERGE, LIST, ALTER, REVOKE, LOAD, UNSET, REMOVE, MOVE}
It costs 0.302s
```

**Comment by Genius-pig** [comment]
@liutaohua 
please be cautious, if you input two many wrong SQL, the parse speed will be slower.

**Pull request merged**